google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageReadChannel.java

```diff
@@ -135,9 +140,31 @@ public int read(ByteBuffer dst) throws IOException {
     }
   }
 
+  private static boolean isReopenable(Throwable exs) {
+    Throwable throwable = exs;
+    // ensures finite iteration
+    int maxDepth = 10;
+    while (throwable != null && maxDepth-- > 0) {
+      Throwable cause = throwable.getCause();
+      if (throwable.getMessage().contains("Connection closed prematurely")
+          || cause instanceof SSLException
+          || cause instanceof EOFException
+          || cause instanceof SocketException
+          || cause instanceof SocketTimeoutException) {
+        return true;
+      }
+      throwable = cause;
+    }
+    return false;
+  }
+
   private void sleepForAttempt(int attempt) {
+    long firstdelay = 1000;
+    long maxDelay = 120_000;
+    // exponential backoff
+    long delay = firstdelay * Math.min(1 << attempt, maxDelay);
```

Comment:
Sorry, this is not exactly what I suggested. If `Math.min(1 << attempt, maxDelay)` returns maxDelay (120,000), then `delay = 1000 * 120,000 = 120,000,000 milliseconds = 33.33 hours (!)`. 

If you need delay to be bounded by maxDelay, then:
```java
long delay = 1000L * (1L << Math.min(7, attempt)); // /2 ^ 7 = 128, closest to 120, so the maximum resulting delay will be 128_000, or 1000 * 2 ^ 7
```
note, the `firstdelay` and `maxDelay` seem redundant (also, pelase note that in the current code they are non-final constants).

or, if you really-really need `maxDelay` to be exactly 120_000:

```java
long delay = 1000L * (1L << Math.min(20, attempt));  //we need to bound power to some reasonable limit, to avoid the shift overflowing number of bits in long
delay = Math.min(delay, 120_000L);
```